app/controllers/characters_controller.rb

```diff
@@ -1,17 +1,21 @@
 class CharactersController < ApplicationController
-  before_action :authenticate_user!
+  protect_from_forgery prepend: true
+  before_action :authenticate_user!, except: [:increment_rank, :decrement_rank, :skills_select]
   before_action :set_character, only: [:edit, :update, :show, :destroy]
   before_action :verify_character, only: [:edit, :update, :show, :destroy]
 
+  def skills_select
+    @skills = Career.find_by(name: params[:career]).career_skills.map { |skill_name| Skill.find_by(name: skill_name) }.compact
```

Comment:
Yes you can, by the end of the day you just need the `career_skills`, so you probably can get rid of the `map`